src/Atc.Azure.Messaging/ServiceBus/ServiceBusCredentialsClientFactory.cs

```diff
@@ -0,0 +1,31 @@
+using Atc.Azure.Options.Authorization;
+using Atc.Azure.Options.Environment;
+using Atc.Azure.Options.Providers;
+using Atc.Azure.Options.ServiceBus;
+using Azure.Identity;
+using Azure.Messaging.ServiceBus;
+
+namespace Atc.Azure.Messaging.ServiceBus;
+
+public class ServiceBusCredentialsClientFactory : IServiceBusClientFactory
```

Comment:
We can consider to be more strict and only expose interfaces, keeping implementation classes `internal`

Reply:
done